src/lib/GovernorTimelockCompound.sol

```diff
@@ -0,0 +1,257 @@
+// PoolTogether Timelock interface:
+// https://etherscan.io/address/0xB3a87172F555ae2a2AB79Be60B336D2F7D0187f0#code#F1#L306
+// Governor Alpha Timelock interface:
+// https://github.com/compound-finance/compound-protocol/blob/a3214f67b73310d547e00fc578e8355911c9d376/contracts/Governance/GovernorAlpha.sol#L320
+
+pragma solidity ^0.8.0;
+
+import {Governor} from "flexible-voting/src/GovernorCountingFractional.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {IGovernorTimelock} from
+  "@openzeppelin/contracts/governance/extensions/IGovernorTimelock.sol";
+import {IPoolTogetherTimelock} from "src/interfaces/IPoolTogetherTimelock.sol";
+import {Timers} from "@openzeppelin/contracts/utils/Timers.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {IERC165} from "@openzeppelin/contracts/interfaces/IERC165.sol";
+import {IGovernor} from "@openzeppelin/contracts/governance/IGovernor.sol";
+
+/**
+ * @dev Extension of {Governor} that binds the execution process to a Compound Timelock. This adds a
+ * delay, enforced by
+ * the external timelock to all successful proposal (in addition to the voting duration). The
+ * {Governor} needs to be
+ * the admin of the timelock for any operation to be performed. A public, unrestricted,
+ * {GovernorTimelockCompound-__acceptAdmin} is available to accept ownership of the timelock.
+ *
+ * Using this model means the proposal will be operated by the {TimelockController} and not by the
+ * {Governor}. Thus,
+ * the assets and permissions must be attached to the {TimelockController}. Any asset sent to the
+ * {Governor} will be
+ * inaccessible.
+ *
+ * _Available since v4.3._
+ */
+abstract contract GovernorTimelockCompound is IGovernorTimelock, Governor {
+  using SafeCast for uint256;
+  using Timers for Timers.Timestamp;
+
+  struct ProposalTimelock {
+    Timers.Timestamp timer;
+  }
+
+  // The interface for this variable was changed to conform to the PoolTogether Timelock interface.
+  //
+  // Original openzeppelin:
+  // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/49c0e4370d0cc50ea6090709e3835a3091e33ee2/contracts/governance/extensions/GovernorTimelockCompound.sol#L31
+  IPoolTogetherTimelock private _timelock;
+
+  mapping(uint256 => ProposalTimelock) private _proposalTimelocks;
+
+  /**
+   * @dev Emitted when the timelock controller used for proposal execution is modified.
+   */
+  event TimelockChange(address oldTimelock, address newTimelock);
+
+  /**
+   * @dev Set the timelock.
+   */
+  // The timelock interface was changed from the original Openzeppelin source to conform to the
+  // PoolTogether interface.
+  //
+  // Original Openzeppelin source:
+  // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/49c0e4370d0cc50ea6090709e3835a3091e33ee2/contracts/governance/extensions/GovernorTimelockCompound.sol#L43
+  constructor(IPoolTogetherTimelock timelockAddress) {
+    _updateTimelock(timelockAddress);
+  }
```

Comment:
In various places we change from natspec comments to regular comments, meaning tools like `forge doc` will likely not match up the natspec with the method. There's a few places in this file to fix this
```suggestion
  /**
   * @dev Set the timelock.
   * The timelock interface was changed from the original Openzeppelin source to conform to the
   * PoolTogether interface.
   *
   * Original Openzeppelin source:
   * https://github.com/OpenZeppelin/openzeppelin-contracts/blob/49c0e4370d0cc50ea6090709e3835a3091e33ee2/contracts/governance/extensions/GovernorTimelockCompound.sol#L43
   */
  constructor(IPoolTogetherTimelock timelockAddress) {
    _updateTimelock(timelockAddress);
  }
```

Reply:
https://github.com/ScopeLift/pooltogether-governor-upgrade/pull/20/commits/7445ab9b7d3b04e10af14887db666cd78b0852f5